src/Widgets/NotificationEntry.vala

```diff
@@ -158,17 +167,24 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
 
         deck.notify["visible-child"].connect (() => {
             if (deck.transition_running == false && deck.visible_child != grid) {
-                remove_notification_entry ();
+                clear ();
             }
         });
 
         deck.notify["transition-running"].connect (() => {
             if (deck.transition_running == false && deck.visible_child != grid) {
-                remove_notification_entry ();
+                clear ();
             }
         });
     }
 
+    public void dismiss () {
+        revealer.reveal_child = false;
+        revealer.notify["child-revealed"].connect (() => {
+            destroy ();
```

Comment:
Generally it's a good idea to check the status here, although I don't think there are any cases here specifically where it can change to revealed status.